torch/distributed/rpc/backend_registry.py

```diff
@@ -152,55 +152,55 @@ def _process_group_init_backend_handler(
     _process_group_init_backend_handler,
 )
 
-def _tensorpipe_construct_rpc_backend_options_handler(
-    rpc_timeout,
-    init_method,
-    num_worker_threads=rpc_constants.DEFAULT_NUM_WORKER_THREADS,
-    _transports=None,
-    _channels=None,
-    **kwargs
-):
-    from . import TensorPipeRpcBackendOptions
-
-    return TensorPipeRpcBackendOptions(
-        rpc_timeout=rpc_timeout,
-        init_method=init_method,
-        num_worker_threads=num_worker_threads,
-        _transports=_transports,
-        _channels=_channels,
-    )
+if sys.platform != 'win32':
```

Comment:
Argh, this is something I forgot in #42682... This check though doesn't have to be Windows-specific. We could do something like:
```
try:
    from . import TensorPipeAgent, TensorPipeRpcBackendOptions
    _USE_TENSORPIPE = True
except ImportError:
    _USE_TENSORPIPE = False

if _USE_TENSORPIPE:
    ...
```
So that this would also cover those people on Linux and Mac who built with USE_TENSORPIPE=0